### ISSUE TITLE
`prefer-robust-stmts` rule does not check DROP INDEX for IF EXISTS clause

### ISSUE THREAD

**Issue by armenzg** [sbdchd/squawk#900]
## Description

The `prefer-robust-stmts` rule currently validates that `DROP TABLE` and `ALTER TABLE DROP COLUMN` statements include `IF EXISTS` clauses, but it does **not** check `DROP INDEX` statements. This creates an inconsistency where migrations can still fail on fresh databases when dropping indexes.

## Current Behavior

```sql
DROP INDEX ix_test;
```

**Squawk output:**
- ✅ Warns about missing `CONCURRENTLY` (via `require-concurrent-index-deletion`)
- ✅ Warns about missing timeout settings
- ❌ **Does NOT** warn about missing `IF EXISTS`

## Expected Behavior

```sql
DROP INDEX ix_test;
```

Should produce:
```
warning[prefer-robust-stmts]: Missing `IF EXISTS`, the migration can't be rerun if it fails part way through.
```

With suggested fix:
```sql
DROP INDEX IF EXISTS ix_test;
```

## Test Case

```bash
# This produces prefer-robust-stmts warning ✅
echo "DROP TABLE test_table;" | squawk

# This produces prefer-robust-stmts warning ✅
echo "ALTER TABLE foo DROP COLUMN bar;" | squawk

# This does NOT produce prefer-robust-stmts warning ❌
echo "DROP INDEX ix_test;" | squawk
```

## Impact

Without this check, migrations can fail in several scenarios:
- Fresh database setup (index never existed)
- Inconsistent development database state
- Out-of-order migration application
- Database reset/recovery scenarios

This is particularly problematic because `DROP INDEX` operations are common in schema evolution, and the failure mode is identical to `DROP TABLE` - yet only one is checked.

## Workaround

We've implemented a custom pre-commit hook to catch this at the Alembic Python API level, but it would be beneficial to have this checked at the SQL level by Squawk for completeness.
https://github.com/getsentry/seer/pull/4677

## Context

- Squawk version: 2.37.0
- Database: PostgreSQL (though this applies to all databases supporting IF EXISTS)
- The `require-concurrent-index-deletion` rule already validates DROP INDEX operations, so the rule detection infrastructure is in place

**Comment by sbdchd** [comment]
I think this is because we have a check to exclude prefer-robust-stmts if it's only one statement.



```bash
❮ echo 'drop index foo;' | cargo run -- --exclude=require-timeout-settings          
warning[require-concurrent-index-deletion]: A normal `DROP INDEX` acquires an `ACCESS EXCLUSIVE` lock on the table, blocking other accesses until the index drop can complete.
  ╭▸ stdin:1:1
  │
1 │ drop index foo;
  │ ━━━━━━━━━━━━━━
  │
  ├ help: Drop the index `CONCURRENTLY`.
  ╭╴
1 │ drop index concurrently foo;
  ╰╴           ++++++++++++

Find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
Found 1 issue in 1 file (checked 1 source file)
```

If I instead add a second statement, then I see:


```bash
❮ echo 'drop index foo; select;' | cargo run -- --exclude=require-timeout-settings
    Finished `dev` profile [unoptimized] target(s) in 0.28s
     Running `target/debug/squawk --exclude=require-timeout-settings`
warning[prefer-robust-stmts]: Missing `IF EXISTS`, the migration can't be rerun if it fails part way through.
  ╭▸ stdin:1:1
  │
1 │ drop index foo; select;
  │ ━━━━━━━━━━━━━━
  ╭╴
1 │ drop index if exists foo; select;
  ╰╴           +++++++++
warning[require-concurrent-index-deletion]: A normal `DROP INDEX` acquires an `ACCESS EXCLUSIVE` lock on the table, blocking other accesses until the index drop can complete.
  ╭▸ stdin:1:1
  │
1 │ drop index foo; select;
  │ ━━━━━━━━━━━━━━
  │
  ├ help: Drop the index `CONCURRENTLY`.
  ╭╴
1 │ drop index concurrently foo; select;
  ╰╴           ++++++++++++

Find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
Found 2 issues in 1 file (checked 1 source file)
```




Although I'm not sure this single statement check still makes sense. I wonder if we're handling an edge case that's not really necessary. What do you think?

**Comment by sbdchd** [comment]
Fixed on latest, thanks for the report!

```
❮ echo 'DROP INDEX ix_test;' | npx squawk-cli@latest --exclude=require-timeout-settings,require-concurrent-index-deletion                         
warning[prefer-robust-stmts]: Missing `IF EXISTS`, the migration can't be rerun if it fails part way through.
  ╭▸ stdin:1:1
  │
1 │ DROP INDEX ix_test;
  │ ━━━━━━━━━━━━━━━━━━
  ╭╴
1 │ DROP INDEX if exists ix_test;
  ╰╴           +++++++++

Find detailed examples and solutions for each rule at https://squawkhq.com/docs/rules
Found 1 issue in 1 file (checked 1 source file)
```
